radixdlt/src/main/java/com/radixdlt/utils/MemoryLeakDetector.java

```diff
@@ -0,0 +1,87 @@
+package com.radixdlt.utils;
+
+import com.sun.management.GarbageCollectionNotificationInfo;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.management.Notification;
+import javax.management.NotificationEmitter;
+import javax.management.NotificationListener;
+import javax.management.openmbean.CompositeData;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MemoryLeakDetector {
+
+    private static final Logger log = LogManager.getLogger();
+
+    private static final String OldGenGcName = "G1 Old Gen";
+
+    private static final int CONSECUTIVE_INCREASE_ALERT_THRESHOLD = 3;
+
+    private final List<Long> memTrail = new ArrayList<>();
```

Comment:
:+1: I'll use LinkedList.
I've considered EvictingQueue but queues don't provide by-index access.

Reply:
Right, but it would be better to use an iterators-based ordering check rather than index-based.

Guava has `Comparators.isInStrictOrder(memTrail, Long::compare)` that gives the right answer and is iterator/iterable based.

Reply:
Nice, I didn't know about that one. I've used it (together with EvictingQueue).